internal/filter/parser_test.go

```diff
@@ -62,101 +62,98 @@ func TestFilter(t *testing.T) {
 	t.Run("ParserIdentifiesAllKindOfFilters", func(t *testing.T) {
 		rule, err := Parse("foo=bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &Equal{}, rule)
+		expected := &Condition{op: Equal, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo!=bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &UnEqual{}, rule)
+		expected = &Condition{op: UnEqual, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo=bar*")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &Like{}, rule)
+		expected = &Condition{op: Like, column: "foo", value: "bar*"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo!=bar*")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &Unlike{}, rule)
+		expected = &Condition{op: UnLike, column: "foo", value: "bar*"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo<bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &LessThan{}, rule)
+		expected = &Condition{op: LessThan, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo<=bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &LessThanOrEqual{}, rule)
+		expected = &Condition{op: LessThanEqual, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo>bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &GreaterThan{}, rule)
+		expected = &Condition{op: GreaterThan, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo>=bar")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &GreaterThanOrEqual{}, rule)
+		expected = &Condition{op: GreaterThanEqual, column: "foo", value: "bar"}
+		assert.Equal(t, expected, rule)
 
 		rule, err = Parse("foo=bar&bar=foo")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &All{}, rule)
+		assert.IsType(t, &Chain{}, rule)
 
 		rule, err = Parse("foo=bar|bar=foo")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &Any{}, rule)
+		assert.IsType(t, &Chain{}, rule)
 
 		rule, err = Parse("!(foo=bar|bar=foo)")
 		assert.Nil(t, err, "There should be no errors but got: %s", err)
-		assert.IsType(t, &None{}, rule)
+		assert.IsType(t, &Chain{}, rule)
```

Comment:
These should now also check `rule.op`.

Reply:
Just ignore theses! There's an extended test case in the other PR.